code/datums/cleaning.dm

```diff
@@ -0,0 +1,60 @@
+//For handling standard click-to-clean items like soap and mops.
+
+//text1, text2, and text3 carry strings that, when pieced together by this proc, make the cleaning messages.
+//text1 and text2 pertain to the start message, i.e. "User begins to [clean] the floor[ with the mop.]"
+//text3 pertains to the end message, i.e. "You [finished mopping] the floor."
+
+/atom/proc/cleaning_act(atom/target, mob/user, atom/cleaner, cleanspeed = 50, text1 = "clean", text2 = " with [src].", text3 = "clean")
```

Comment:
Can you rewrite this proc without the target var? The target should be `src`, so we can use that instead.